packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -865,135 +906,141 @@ export function hydrateHoistable(
+      // case 'rowSpan':
+      // case 'start':
+
+      default:
+        if (isAttributeNameSafe(propName)) {
           if (__DEV__) {
             checkAttributeStringCoercion(propValue, propName);
           }
-          if (node.getAttribute(propName) !== '' + propValue) continue nodeLoop;
-          break;
+          attributes.push(propName, '' + propValue);
         }
+    }
+  }
 
-        // Boolean props used by hoistable tag types
-        case 'async':
-        case 'defer':
-        case 'disabled':
-        case 'hidden':
-        case 'noModule':
-        case 'scoped':
-        case 'itemScope':
-          if (propValue !== node.hasAttribute(propName)) continue nodeLoop;
-          break;
-
-        // The following properties are left out because they do not apply to
-        // the current set of hoistable types. They may have special handling
-        // requirements if they end up applying to a hoistable type in the future
-        // case 'acceptCharset':
-        // case 'value':
-        // case 'allowFullScreen':
-        // case 'autoFocus':
-        // case 'autoPlay':
-        // case 'controls':
-        // case 'default':
-        // case 'disablePictureInPicture':
-        // case 'disableRemotePlayback':
-        // case 'formNoValidate':
-        // case 'loop':
-        // case 'noValidate':
-        // case 'open':
-        // case 'playsInline':
-        // case 'readOnly':
-        // case 'required':
-        // case 'reversed':
-        // case 'seamless':
-        // case 'multiple':
-        // case 'selected':
-        // case 'capture':
-        // case 'download':
-        // case 'cols':
-        // case 'rows':
-        // case 'size':
-        // case 'span':
-        // case 'rowSpan':
-        // case 'start':
-
-        default:
-          if (isAttributeNameSafe(propName)) {
-            const attributeName = propName;
-            if (propValue == null && node.hasAttribute(attributeName))
-              continue nodeLoop;
-            if (__DEV__) {
-              checkAttributeStringCoercion(propValue, attributeName);
-            }
-            if (node.getAttribute(attributeName) !== '' + (propValue: any))
-              continue nodeLoop;
-          }
-      }
-      checkedAttributes++;
+  const nodesIter = hoistedTags.values();
+  let step = nodesIter.next();
+  nodeLoop: for (; !step.done; step = nodesIter.next()) {
+    const instance = step.value;
+    if (instance.textContent !== childString) {
+      continue;
     }
 
-    if (node.attributes.length !== checkedAttributes) {
-      // We didn't match ever attribute so we abandon this node
-      continue nodeLoop;
+    // attributes uses two slots per attribute. We normalize
+    // on double length to test whether we have the right number of attributes
+    if (instance.attributes.length * 2 !== attributes.length) {
+      // This node has a different number of attributes than our instance expects
+      continue;
```

Comment:
one crazy Idea I had is to cycle the elements in the Set so that once we have found a match the first value is whatever is next in the iterator. The rationale is that for serial resources the matching will happen without any iteration unless there is no match to be found at all. I'm guess this is probably actually much slower though b/c the extra mutation on every step so I didn't...